sdk/core/azure-core/inc/logging/logging.hpp

```diff
@@ -0,0 +1,69 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#pragma once
+
+#include "azure.hpp"
+
+#include <algorithm>
+#include <functional>
+#include <set>
+#include <string>
+
+namespace Azure { namespace Core { namespace Logging {
+  class LogClassification;
+
+  typedef std::function<
+      void(LogClassification const& classification, std::string const& message) noexcept>
+      LogListener;
+
+  void SetLogListener(LogListener logListener);
+  void ResetLogListener();
```

Comment:
We really can't get rid of this function and instead pass null to SetLogListener?

Reply:
We can - both `nullptr` and `Reset()` will have the same effect. Should I remove `Reset()`?

Also, regarding classifications. I can add `LogClassification::All` (or `LogClassification::Everything`) and make it to behave in a way so that `SetClassifications(LogClassification::Everything)` behaves like `ResetLogClassifications()`. Would you like to see that?

Another thing (orthogonal to `::Everything` proposal ablove) - I can make classifications to behave symantically like bit fields - so that you can combine them with `|` and so on. That would require some header code; it will be mostly hidden in the `Details` namespace however. Would you like that? I assume no, but I am listing all options for you here. I remember Rick wanted something like that. BTW, we can add that later, without breaking compile-time API.

Reply:
I added `LogClassification::All` and `LogClassification::None`, and got rid of both Resets().